includes/Checker/Checks/File_Type_Check.php

```diff
@@ -0,0 +1,184 @@
+	 *
+	 * @param int $flags Bitwise flags to control check behavior.
+	 */
+	public function __construct( $flags = self::TYPE_ALL ) {
+		$this->flags = $flags;
+	}
+
+	/**
+	 * Amends the given result by running the check on the given list of files.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param Check_Result $result The check result to amend, including the plugin context to check.
+	 * @param array        $files  List of absolute file paths.
+	 *
+	 * @throws Exception Thrown when the check fails with a critical error (unrelated to any errors detected as part of
+	 *                   the check).
+	 */
+	protected function check_files( Check_Result $result, array $files ) {
+		if ( $this->flags & self::TYPE_COMPRESSED ) {
+			$this->look_for_compressed_files( $result, $files );
+		}
+		if ( $this->flags & self::TYPE_PHAR ) {
+			$this->look_for_phar_files( $result, $files );
+		}
+		if ( $this->flags & self::TYPE_VCS ) {
+			$this->look_for_vcs_directories( $result, $files );
+		}
+		if ( $this->flags & self::TYPE_HIDDEN ) {
+			$this->look_for_hidden_files( $result, $files );
+		}
+	}
+
+	/**
+	 * Looks for compressed files and amends the given result with an error if found.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param Check_Result $result The check result to amend, including the plugin context to check.
+	 * @param array        $files  List of absolute file paths.
+	 */
+	protected function look_for_compressed_files( Check_Result $result, array $files ) {
+		$compressed_files = self::filter_files_by_extensions( $files, array( 'zip', 'gz', 'tgz', 'rar', 'tar', '7z' ) );
+		if ( $compressed_files ) {
+			foreach ( $compressed_files as $file ) {
+				$result->add_message(
+					true,
+					'Compressed files are not permitted.',
+					array(
+						'code' => 'compressed_files',
+						'file' => str_replace( $result->plugin()->path(), '', $file ),
+					)
+				);
+			}
+		}
+	}
+
+	/**
+	 * Looks for PHAR files and amends the given result with an error if found.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param Check_Result $result The check result to amend, including the plugin context to check.
+	 * @param array        $files  List of absolute file paths.
+	 */
+	protected function look_for_phar_files( Check_Result $result, array $files ) {
+		$phar_files = self::filter_files_by_extension( $files, 'phar' );
+		if ( $phar_files ) {
+			foreach ( $phar_files as $file ) {
+				$result->add_message(
+					true,
+					'Phar files are not permitted.',
+					array(
+						'code' => 'phar_files',
+						'file' => str_replace( $result->plugin()->path(), '', $file ),
+					)
+				);
+			}
+		}
+	}
+
+	/**
+	 * Looks for VCS directories and amends the given result with an error if found.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param Check_Result $result The check result to amend, including the plugin context to check.
+	 * @param array        $files  List of absolute file paths.
+	 */
+	protected function look_for_vcs_directories( Check_Result $result, array $files ) {
```

Comment:
Not, I have used composer to deploy WP + plugins to production. If composer does a git checkout, this would result in .git files in production.

Reply:
This is taken over from the plugin review team's tool, so I think we should stick with that for now for parity. If this becomes a concern, we can further down in the future report an issue, but let's not do that just yet.